### ISSUE TITLE
class decorators not possible to type hint

### ISSUE THREAD

**Issue by Rapptz** [microsoft/pyright#1718]
**Describe the bug**
A class decorator is a decorator that takes in `cls` and then returns something (usually `cls` again, after it's been augmented). I wanted to type hint this but got errors in pylance/pyright.

**To Reproduce**
Make a class decorator and type hint it

```py
from typing import Callable, Type, Any

def foo() -> Callable[[Type[Any]], Type[Any]]:
  def dec(cls: Type[Any]) -> Type[Any]:
    return cls
  return dec
```

**Expected behavior**
Pass the type check

**Screenshots or Code**
![image](https://user-images.githubusercontent.com/1695103/113420221-eddd1a80-9396-11eb-8bed-8b98df6b7413.png)

**VS Code extension or command-line**
VS Code extension. Pylance version 2021.3.4. Sorry if I'm reporting this in the wrong place.

**Additional Context**

This type checks fine on mypy.

For some reason, using a `TypeVar` in place of `Any` makes it type check, i.e. this passes

```py
_T = TypeVar('_T')

def foo() -> Callable[[Type[_T]], Type[_T]]:
  def dec(cls: Type[_T]) -> Type[_T]:
    return cls
  return dec
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This fix is included in pyright 1.1.128, which I just published. It will also be included in the next release of pylance.
